gradle-plugins/aot-smoke-test-plugin/src/main/java/org/springframework/aot/gradle/AotSmokeTestPlugin.java

```diff
@@ -105,8 +107,11 @@ private void configureJvmTests(Project project, SourceSet aotTest, AotSmokeTestE
 
 	private void configureNativeImageTests(Project project, SourceSet aotTest, AotSmokeTestExtension extension) {
 		project.getPlugins().withType(NativeImagePlugin.class, (nativeImagePlugin) -> {
-			project.getExtensions().getByType(GraalVMExtension.class).getAgent().getTasksToInstrumentPredicate()
-					.set((task) -> false);
+			GraalVMExtension graalVMExtension = project.getExtensions().getByType(GraalVMExtension.class);
+			graalVMExtension.getAgent().getTasksToInstrumentPredicate().set((task) -> false);
+			((ExtensionAware) graalVMExtension).getExtensions()
+					.getByType(GraalVMReachabilityMetadataRepositoryExtension.class)
+					.getVersion().set((String) project.getProperties().get("reachabilityMetadataVersion"));
```

Comment:
Shouldn't we allow to use any URL we want? This way we could make testing of previews more easy.

Reply:
I pushed an updated commit with support for both version and URL.